proxy.go

```diff
@@ -107,6 +107,8 @@ func (s *validTailnetSrv) rewrite(r *httputil.ProxyRequest) {
 			r.Out.Header.Set("X-Forwarded-Server", hostOnly)
 			r.Out.Header.Set("X-Forwarded-Port", port)
 		}
+    r.Out.Header.Set("Upgrade", r.In.Header.Get("Upgrade"))
+    r.Out.Header.Set("Connection", "upgrade")
```

Comment:
Hm, maybe we can set that from the input Connection header as well? Seems unfortunate to unconditionally upgrade connections if downstream client expects them to be closed.